### PR TITLE
Convert to markdown: named_tensor.rst, nested.rst, nn.attention.bias.rst, nn.attention.experimental.rst, nn.attention.flex_attention.rst #155028

### LINKED ISSUE #155028: Convert to markdown: named_tensor.rst, nested.rst, nn.attention.bias.rst, nn.attention.experimental.rst, nn.attention.flex_attention.rst
Convert the following files to MyST markdown and ensure that the doc tests are green:

- [named_tensor.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/named_tensor.rst)
- [nested.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/nested.rst)
- [nn.attention.bias.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/nn.attention.bias.rst)
- [nn.attention.experimental.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/nn.attention.experimental.rst)
- [nn.attention.flex_attention.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/nn.attention.flex_attention.rst)

To convert an `.rst` file:

1) Assign the issue to yourself using the `/assigntome` comment.

2) **Important!** Use `git mv filename.rst filename.md` to convert the issue from `.rst` to `.md`. **DO NOT delete**
   the `.rst` and create the `.md` - we won't accept PRs with deleted `.rst` files. We want to preserve the git log of
   those files. Therefore, using `git mv` is very important!

3) Before submitting a PR, build the docs locally as described in the [README](https://github.com/pytorch/pytorch/blob/main/README.md#building-the-documentation).
   Preview localy with `make serve` to ensure that the HTML renders correctly.    

4) Run `lintrunner` from the root of the repository:

   ```
   pip install lintrunner
   lintrunner init
   lintrunner -m main
   ```

4) Submit a PR with your changes and ensure the PR passes the doc CI tests. Specifically it needs to pass:

   * pull / linux-docs / build-docs-python-false (pull_request)
   * linux-jammy-py3.9-gcc11 / test (docs_test, 1, 1, lf.linux.2xlarge)

   Doc tests start after the Docker build, which might take about 30 minutes. 
   The doc build itself should take about 25 minutes. When the build is done, 
   you can see the preview of your build by clicking:
   📄 **Preview Python docs built from this PR** in the `pytorch-bot` comment.

   Make sure your PR is tagged with `docathon-h1-2025`, `module: docs`, `topic: not user facing`,
   and `topic: docs` labels.

5) Compare the preview of the pages you converted with same pages it the main branch.
   The converted pages in preview must look the same as in the main branch.

6) Once the PR is accepted, one of the maintainers will merge it.

See [Quick MyST Conversion Reference](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines#restructuredtext-vs-markdown)
in our wiki or [MyST Parser documentation](https://myst-parser.readthedocs.io/en/v0.15.1/sphinx/use.html) for detailed syntax.

Here is an example [PR](https://github.com/pytorch/pytorch/pull/154438)

cc @sekyondaMeta @AlannaBurke

### PR DESCRIPTION
**Pull request by eromomon**
Fixes #155028 

This pull request updates the documentation  by transitioning from .rst to .md format. It introduces new Markdown files for the documentation of named_tensor, nested, nn.attention.bias, nn.attention.experimental, and nn.attention.flex_attention

cc @svekars @sekyondaMeta @AlannaBurke

### CODE REVIEW

File: docs/source/named_tensor.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
+```{eval-rst}
 .. currentmodule:: torch
+```
 
-.. _named_tensors-doc:
-
-Named Tensors
-=============
+(named_tensors-doc)=
```
Comment (svekars) [inline docs/source/named_tensor.md:5]:
```suggestion
(named_tensors-doc)=

```

File: docs/source/named_tensor.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
+## Named Tensors
```
Comment (svekars) [inline docs/source/named_tensor.md]: should be heading 1
```suggestion
# Named Tensors
```

File: docs/source/named_tensor.md [hunk 3]
```diff
@@ -14,64 +14,62 @@ also be used to rearrange dimensions, for example, to support
...
 
 
-.. warning::
+```{warning}
     The named tensor API is a prototype feature and subject to change.
+```
 
-Creating named tensors
-----------------------
+### Creating named tensors
```
Comment (svekars) [inline docs/source/named_tensor.md]:
```suggestion
## Creating named tensors
```

File: docs/source/named_tensor.md [hunk 4]
```diff
@@ -14,64 +15,62 @@ also be used to rearrange dimensions, for example, to support
...
-----------------
+- {func}`torch.empty`
+- {func}`torch.rand`
+- {func}`torch.randn`
+- {func}`torch.ones`
+- {func}`torch.tensor`
+- {func}`torch.zeros`
 
-See :attr:`~Tensor.names` for restrictions on tensor names.
+### Named dimensions
```
Comment (svekars) [inline docs/source/named_tensor.md]:
```suggestion
## Named dimensions
```

File: docs/source/named_tensor.md [hunk 5]
```diff
@@ -14,64 +15,62 @@ also be used to rearrange dimensions, for example, to support
...
-
+```
     >>> imgs = torch.randn(1, 2, 2, 3 , names=(None, 'C', 'H', 'W'))
     >>> imgs.names
     (None, 'C', 'H', 'W')
+```
 
-Name propagation semantics
---------------------------
+### Name propagation semantics
```
Comment (svekars) [inline docs/source/named_tensor.md]:
```suggestion
## Name propagation semantics
```

File: docs/source/named_tensor.md [hunk 6]
```diff
@@ -83,17 +82,16 @@ More formally, name inference consists of the following two steps:
...
+```
     >>> x = torch.randn(3, 3, names=('N', 'C'))
     >>> x.abs().names
     ('N', 'C')
+```
 
 
-.. _match_semantics-doc:
+(match_semantics-doc)=
+#### match semantics
```
Comment (svekars) [inline docs/source/named_tensor.md]:
```suggestion
### match semantics
```

File: docs/source/named_tensor.md [hunk 7]
```diff
@@ -102,62 +100,59 @@ Nones are essentially a special "wildcard" name.
...
     In practice, when working with named tensors, one should avoid having unnamed
     dimensions because their handling can be complicated. It is recommended to lift
-    all unnamed dimensions to be named dimensions by using :meth:`~Tensor.refine_names`.
-
+    all unnamed dimensions to be named dimensions by using {meth}`~Tensor.refine_names`.
+```
 
-Basic name inference rules
-^^^^^^^^^^^^^^^^^^^^^^^^^^
+#### Basic name inference rules
```
Comment (svekars) [inline docs/source/named_tensor.md]:
```suggestion
### Basic name inference rules
```

File: docs/source/named_tensor.md [hunk 8]
```diff
@@ -102,62 +100,59 @@ Nones are essentially a special "wildcard" name.
...
 Here are two common operations that may be useful to go over:
 
-- Binary arithmetic ops: :ref:`unifies_names_from_inputs-doc`
-- Matrix multiplication ops: :ref:`contracts_away_dims-doc`
+- Binary arithmetic ops: {ref}`unifies_names_from_inputs-doc`
+- Matrix multiplication ops: {ref}`contracts_away_dims-doc`
 
-Explicit alignment by names
----------------------------
+### Explicit alignment by names
```
Comment (svekars) [inline docs/source/named_tensor.md]:
```suggestion
## Explicit alignment by names
```

File: docs/source/named_tensor.md [hunk 9]
```diff
@@ -173,29 +168,29 @@ by name to a specified ordering. This is useful for performing "broadcasting by
...
     >>> scale_channels(more_imgs, scale)
     >>> scale_channels(videos, scale)
+```
 
-Manipulating dimensions
------------------------
-
-Use :meth:`~Tensor.align_to` to permute large amounts of dimensions without
-mentioning all of them as in required by :meth:`~Tensor.permute`.
+### Manipulating dimensions
```
Comment (svekars) [inline docs/source/named_tensor.md]:
```suggestion
## Manipulating dimensions
```

File: docs/source/named_tensor.md [hunk 10]
```diff
@@ -207,18 +202,16 @@ and :meth:`~Tensor.reshape`, but have more semantic meaning to someone reading t
...
     >>> unflattened_named_imgs.names
     ('N', 'C', 'H', 'W')
+```
 
-.. _named_tensors_autograd-doc:
-
-Autograd support
-----------------
+(named_tensors_autograd-doc)=
+### Autograd support
```
Comment (svekars) [inline docs/source/named_tensor.md]:
```suggestion
## Autograd support
```

File: docs/source/named_tensor.md [hunk 11]
```diff
@@ -234,31 +227,30 @@ safety that names give us.
     >>> loss.backward(grad_loss)
     >>> weight.grad
     tensor([-1.8107, -0.6357,  0.0783])
+```
 
-Currently supported operations and subsystems
----------------------------------------------
+### Currently supported operations and subsystems
```
Comment (svekars) [inline docs/source/named_tensor.md]:
```suggestion
## Currently supported operations and subsystems
```

File: docs/source/named_tensor.md [hunk 12]
```diff
@@ -234,31 +227,30 @@ safety that names give us.
...
 
-Operators
-^^^^^^^^^
+#### Operators
```
Comment (svekars) [inline docs/source/named_tensor.md]:
```suggestion
### Operators
```

File: docs/source/named_tensor.md [hunk 13]
```diff
@@ -234,31 +227,30 @@ safety that names give us.
...
+- {func}`torch.nn.functional.relu`
+- {func}`torch.nn.functional.softmax`
+- {func}`torch.nn.functional.log_softmax`
+- {func}`torch.nn.functional.tanh`
+- {func}`torch.nn.functional.sigmoid`
+- {func}`torch.nn.functional.dropout`
 
-Subsystems
-^^^^^^^^^^
+#### Subsystems
```
Comment (svekars) [inline docs/source/named_tensor.md]:
```suggestion
### Subsystems
```

File: docs/source/named_tensor.md [hunk 14]
```diff
@@ -272,34 +264,45 @@ We also do not support the following subsystems, though some may work out
...
 
 If any of these would help your use case, please
-`search if an issue has already been filed <https://github.com/pytorch/pytorch/issues?q=is%3Aopen+is%3Aissue+label%3A%22module%3A+named+tensor%22>`_
-and if not, `file one <https://github.com/pytorch/pytorch/issues/new/choose>`_.
+[search if an issue has already been filed](https://github.com/pytorch/pytorch/issues?q=is%3Aopen+is%3Aissue+label%3A%22module%3A+named+tensor%22)
+and if not, [file one](https://github.com/pytorch/pytorch/issues/new/choose).
 
-Named tensor API reference
---------------------------
+### Named tensor API reference
```
Comment (svekars) [inline docs/source/named_tensor.md]:
```suggestion
## Named tensor API reference
```

File: docs/source/named_tensor.md [hunk 15]
```diff
@@ -272,34 +264,45 @@ We also do not support the following subsystems, though some may work out
...
    .. automethod:: refine_names
-
+```
+```{eval-rst}
    .. automethod:: align_as
+```
+```{eval-rst}
    .. automethod:: align_to
-
+```
```
Comment (svekars) [inline docs/source/named_tensor.md]: I think these should all be under one eval-rst otherwise sphinx [doesn't understand](https://github.com/pytorch/pytorch/actions/runs/15594984953/job/43924502305?pr=155696#step:11:818) where to look for them:
```
```{eval-rst}
.. class:: Tensor()
   :noindex:

   .. autoattribute:: names

   .. automethod:: rename

   .. automethod:: rename_

   .. automethod:: refine_names

   .. automethod:: align_as

   .. automethod:: align_to

```

Comment (eromomon) [reply docs/source/named_tensor.md]: solved

File: docs/source/named_tensor.md [hunk 16]
```diff
@@ -102,62 +100,59 @@ Nones are essentially a special "wildcard" name.
 It returns the more *specific* of the two names, if they match. If the names do not match,
 then it errors.
 
-.. note::
+```{note}
```
Comment (AlannaBurke) [inline docs/source/named_tensor.md:103]: you've got an unneeded code block here: 
Monosnap Named Tensors — PyTorch main documentation 2025-06-13 15-14-14

**Review by AlannaBurke — CHANGES_REQUESTED** [review]
If you fix that code block in the note, this PR will be good.

File: docs/source/named_tensor.md (hunk 16 above)
Comment (eromomon) [reply docs/source/named_tensor.md:103]: Thanks.
Code block removed; it retains the same formatting as in the initial .rst file

File: docs/source/named_tensor.md [hunk 17]
```diff
@@ -102,62 +100,60 @@ Nones are essentially a special "wildcard" name.
 It returns the more *specific* of the two names, if they match. If the names do not match,
 then it errors.
 
-.. note::
-    In practice, when working with named tensors, one should avoid having unnamed
-    dimensions because their handling can be complicated. It is recommended to lift
-    all unnamed dimensions to be named dimensions by using :meth:`~Tensor.refine_names`.
+Note
```
Comment (svekars) [inline docs/source/named_tensor.md]: can you wrap this in a proper note directive: ```{note}

**Review by svekars — COMMENTED** [review]
One small comment to fix. Otherwise, LGTM

File: docs/source/named_tensor.md (hunk 17 above)
Comment (eromomon) [reply docs/source/named_tensor.md]: Solved, note directive applied

**Review by svekars — APPROVED** [review]

**Review by svekars — APPROVED** [review]
LGTM
